pulsar-broker/src/main/java/org/apache/pulsar/broker/service/nonpersistent/NonPersistentTopic.java

```diff
@@ -1119,14 +1128,14 @@ public CompletableFuture<MessageId> getLastMessageId() {
     private static final Logger log = LoggerFactory.getLogger(NonPersistentTopic.class);
 
     @Override
-    public CompletableFuture<Void> addSchemaIfIdleOrCheckCompatible(SchemaData schema) {
+    public CompletableFuture<Void> addSchemaIfIdleOrCheckCompatible(SchemaData schema, String subscriptionName) {
         return hasSchema().thenCompose((hasSchema) -> {
             int numActiveConsumers = subscriptions.values().stream()
                     .mapToInt(subscription -> subscription.getConsumers().size())
                     .sum();
             if (hasSchema
                     || (!producers.isEmpty())
-                    || (numActiveConsumers != 0)
+                    || (numActiveConsumers != 0 && !subscriptionsOnlyIncludedAutoSchema.containsKey(subscriptionName))
```

Comment:
its better to check no schema consumer count and numActiveConsumers. if they have the same number, we add the schema directly, if not same then `checkSchemaCompatibleForConsumer`

Reply:
In addConsumerSchemaTypeForSubscriptions method, `subscriptionsOnlyIncludedAutoSchema`  saves only the subscription that contains only AUTO Schema Consumers. So we check whether subscriptionsOnlyIncludedAutoSchema contains the current subscription.subscriptionsOnlyIncludedAutoSchema saves only the subscription that contains only AUTO Schema Consumers

Reply:
In `addConsumerSchemaTypeForSubscriptions`, the subscriptionsOnlyIncludedAutoSchema saves only the subscription that contains only AUTO schema consumers. So we should check whether subscriptionsOnlyIncludedAutoSchema contains the current subscription and numActiveConsumers. If subscriptionsOnlyIncludedAutoSchema save all subscription and its corresponding all schematype consumer, then we can check the no schema consumer count. But I think it's a waste.

Reply:
1. the root cause is, we can't distinguish no schema or AUTO_CONSUME schema, because the client sub command doesn't upload the AUT0_CONSUME type to broker
2. use a map to check this is very memory consuming, and not necessary